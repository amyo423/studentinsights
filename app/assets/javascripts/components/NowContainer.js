import React from 'react';

// A container for testing that allows passing in a `nowFn`
// that will be shared with child components through context.
export default class NowContainer extends React.Component {
  getChildContext() {
    const {nowFn} = this.props;
    return {
      nowFn() { return nowFn(); }
    };
  }

  render() {
    const {children} = this.props;

    return children;
  }
}
NowContainer.childContextTypes = {
  nowFn: React.PropTypes.func
};
NowContainer.propTypes = {
  children: React.PropTypes.node.isRequired,
  nowFn: React.PropTypes.func.isRequired
};