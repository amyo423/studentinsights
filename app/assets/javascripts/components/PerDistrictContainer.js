import React from 'react';
import PropTypes from 'prop-types';


// A container that provides `districtKey`, assumed to be static across
// the lifetime of all JS execution.  For React components, use this instead of
// `readEnv`, which is intended for bootstrap code.
export default class PerDistrictContainer extends React.Component {
  getChildContext() {
    const {districtKey} = this.props;
    return {districtKey};
  }

  render() {
    const {children} = this.props;
    return children;
  }
}
PerDistrictContainer.childContextTypes = {
  districtKey: PropTypes.string
};
PerDistrictContainer.propTypes = {
  districtKey: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired
};
