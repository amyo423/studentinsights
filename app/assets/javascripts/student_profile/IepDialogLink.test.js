import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {withDefaultNowContext} from '../testing/NowContainer';
import PerDistrictContainer from '../components/PerDistrictContainer';
import IepDialogLink from './IepDialogLink';
import serializedDataForPlutoPoppins from './fixtures/serializedDataForPlutoPoppins.fixture';

function testProps(props = {}) {
  return {
    student: serializedDataForPlutoPoppins.student,
    iepDocument: {
      id: 789
    },
    linkEl: null,
    ...props
  };
}

function testEl(props, context = {}) {
  const districtKey = context.districtKey || 'somerville';
  return withDefaultNowContext(
    <PerDistrictContainer districtKey={districtKey}>
      <IepDialogLink {...props} />
    </PerDistrictContainer>
  );
}

export function renderAcrossAllCombinations() {
  const props = testProps();
  const districtKeys = [
    'somerville',
    'bedford',
    'new_bedford'
  ];

  const els = districtKeys.map(districtKey => (
    <div key={districtKey} style={{margin: 10}}>
      <div>{districtKey}</div>
      <div style={{border: '1px solid #ccc'}}>{testEl(props, {districtKey})}</div>
    </div>
  ));
  return <div>{els}</div>;
}

function snapshotTree(districtKey, props = {}) {
  return renderer
    .create(testEl(testProps(props), {districtKey}))
    .toJSON();
}

it('renders without crashing', () => {
  const el = document.createElement('div');
  const props = testProps();
  ReactDOM.render(testEl(props), el);
});

describe('snapshots', () => {
  it('works for Somerville', () => expect(snapshotTree('somerville')).toMatchSnapshot());
  it('works for Bedford', () => expect(snapshotTree('bedford')).toMatchSnapshot());
  it('works for New Bedford', () => expect(snapshotTree('new_bedford')).toMatchSnapshot());
});
