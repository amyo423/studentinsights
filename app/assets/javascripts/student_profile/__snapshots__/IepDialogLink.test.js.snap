// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots works for Bedford 1`] = `
<span>
  IEP for Communication
</span>
`;

exports[`snapshots works for New Bedford 1`] = `
<div
  style={
    Object {
      "display": "block",
      "marginLeft": 0,
    }
  }
>
  <a
    href="#"
    onClick={[Function]}
    style={
      Object {
        "display": "inline-block",
        "fontSize": 14,
        "outline": "none",
      }
    }
  >
    IEP for Communication
  </a>
</div>
`;

exports[`snapshots works for Somerville 1`] = `
<div
  style={
    Object {
      "display": "block",
      "marginLeft": 0,
    }
  }
>
  <a
    href="#"
    onClick={[Function]}
    style={
      Object {
        "display": "inline-block",
        "fontSize": 14,
        "outline": "none",
      }
    }
  >
    IEP for Communication
  </a>
</div>
`;
